File: web_manager/src/main/webapp/js/controller/orderController.js
//控制层
app.controller("orderController" ,function($scope,$controller,$http ,orderService) {

    $controller('baseController', {$scope: $scope});//继承

    $scope.searchEntity={};

    // 假设定义一个查询的实体：searchEntity
    $scope.search = function(page,rows){
        // 向后台发送请求获取数据:
        orderService.search(page,rows,$scope.searchEntity).success(function(response){
            $scope.paginationConf.totalItems = response.total;
            $scope.list = response.rows;
        });
    }

    //读取列表数据绑定到表单中
    $scope.findAll=function(){
        orderService.findAll().success(
            function(response){
                $scope.list=response;
            }
        );
    }

    $scope.status = ["未付款","已付款","未发货","已发货","交易成功","交易关闭","待评价"];

    //分页
    $scope.findPage=function(page,rows){
        orderService.findPage(page,rows).success(
            function(response){
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }

})

File: web_manager/src/main/java/cn/itcast/core/controller/OrderController.java
package cn.itcast.core.controller;

import cn.itcast.core.pojo.entity.PageResult;
import cn.itcast.core.pojo.good.Brand;
import cn.itcast.core.pojo.order.Order;
import cn.itcast.core.service.OrderService;
import com.alibaba.dubbo.config.annotation.Reference;
import com.github.pagehelper.Page;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/order")
public class OrderController {

    @Reference
    private OrderService orderService;

    @RequestMapping("/findAll")
    public List<Order> findAll(){
        return orderService.findAll();
    }

    /**
     * 高级查询(分页, 高级查询)
     * @param page  当前页
     * @param rows  每页展示多少条数据
     * @param order 需要查询的条件品牌对象
     * @return*/

    @RequestMapping("/search")
    public PageResult search(Integer page, Integer rows,@RequestBody Order order) {
        PageResult pageResult = orderService.findPage(page,rows,order);
        return pageResult;
    }

    @RequestMapping("/findPage")
    public PageResult findPage(Integer page, Integer rows){
        PageResult pageResult = orderService.findPage(page, rows,null);
        return pageResult;
    }
}
